Test/Unit/Model/Backend/QueueCronTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Test\Unit\Model\Backend;
+
+use Algolia\AlgoliaSearch\Exception\InvalidCronException;
+use Algolia\AlgoliaSearch\Model\Backend\QueueCron;
+use Magento\Framework\App\Cache\TypeListInterface;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\Event\ManagerInterface;
+use Magento\Framework\Model\Context;
+use Magento\Framework\Registry;
+use PHPUnit\Framework\TestCase;
+
+class QueueCronTest extends TestCase
+{
+    protected ?QueueCron $queueCronModel;
+
+    protected function setUp(): void
+    {
+        $context = $this->createMock(Context::class);
+        $eventDispatcher = $this->createMock(ManagerInterface::class);
+        $context->method('getEventDispatcher')->willReturn($eventDispatcher);
+
+        $registry = $this->createMock(Registry::class);
+        $config = $this->createMock(ScopeConfigInterface::class);
+        $cacheTypeList = $this->createMock(TypeListInterface::class);
+
+        $this->queueCronModel = new QueueCron($context, $registry, $config, $cacheTypeList);
+    }
+
+    public function testInput(): void
+    {
+        $valuesToTest = [
```

Comment:
I like all of these test cases! I think it might be cleaner to introduce these as a [separate data provider](https://phpunit.de/manual/3.7/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers).